pkg/controller/deployment_inplace.go

```diff
@@ -284,6 +290,13 @@ func (dc *controller) reconcileOldMachineSetsInPlace(ctx context.Context, allMac
 		return false, err
 	}
 
+	// Machines from old machine sets which are undergoing update will eventually move to new machine set.
+	// So once the current new machine set replcas + old machine set replicas undergoing update reaches the desired replicas,
+	// we can stop selecting machines from old machine sets for update.
+	if oldISsMachinesUndergoingUpdate+newMachineSet.Spec.Replicas >= deployment.Spec.Replicas {
```

Comment:
Can you please refrain from using `IS` and instead use `MachineSet`?

Reply:
Adapted it, PTAL